src/controller/metrics.controller.ts

```diff
@@ -0,0 +1,172 @@
+import { PrettyPrintableError } from '@oclif/core/lib/interfaces';
+import { CLIErrorCodes } from '../shared/errorCodes';
+import { factory } from '../service/faasFactory.service';
+import {
+  BUCKET_SIZES as predefinedBucketSizes,
+  THIRTY_DAYS,
+  FIFTEEN_MINUTES,
+} from '../shared/constants';
+import { MetricsView } from '../view/metrics.view';
+
+interface IInputFlags {
+  start?: string | number;
+  end?: string | number;
+  last?: string;
+  output?: string;
+}
+
+interface IMetricsConfig {
+  lambdaFunction: string;
+  inputFlags: IInputFlags;
+}
+
+export class MetricsController {
+  private metricsView: MetricsView;
+
+  constructor({ metricsView = new MetricsView() } = {}) {
+    this.metricsView = metricsView;
+  }
+
+  /**
+   * Gets the logs of the passed function
+   * @param {IMetricsConfig} - lambda function and flags
+   * @returns {Promise<void>}
+   * @memberof MetricsController
+   */
+  public async getMetrics({
+    lambdaFunction,
+    inputFlags,
+  }: IMetricsConfig): Promise<void> {
+    const { start, end = Date.now(), last, output = '' } = inputFlags;
+
+    let startTimestamp: number;
+    const endTimestamp = Number(end);
+
+    if (!start && !last) {
+      throw new Error(
+        'Please define either start and end timestamp or define the period for the last 5m, 1h, 7d,... since now',
+      );
+    }
+
+    if (start && !last) {
+      startTimestamp = Number(start);
+    } else {
+      startTimestamp = MetricsController.calculateStartTimestamp({
+        end,
+        last,
+      });
+    }
+
+    if (Number.isNaN(startTimestamp) || Number.isNaN(endTimestamp)) {
+      throw new Error('Given timestamps are not valid');
+    }
+
+    if (startTimestamp > endTimestamp) {
+      throw new Error('Start timestamp has to be before end timestamp.');
+    }
+
+    if (endTimestamp - startTimestamp > THIRTY_DAYS) {
+      throw new Error('Time period cannot exceed 30 days.');
+    }
+
+    if (endTimestamp - Number(startTimestamp) < FIFTEEN_MINUTES) {
```

Comment:
Why only here the Number casting? Seems 🐟

Reply:
Forgot to remove. Validity is checked before